pyop2/codegen/builder.py

```diff
@@ -442,6 +453,85 @@ def emit_unpack_instruction(self, *, loop_indices=None):
         yield call
 
 
+class MixedMatPack(Pack):
+
+    def __init__(self, packs, access, dtype, block_shape, interior_horizontal=False):
+        self.access = access
+        assert len(block_shape) == 2
+        self.packs = numpy.asarray(packs).reshape(block_shape)
+        self.dtype = dtype
+        self.interior_horizontal = interior_horizontal
```

Comment:
What is `interior_horizontal` exactly? Is this part of the "extrusion hack" in PyOP2 currently? I'm also not quite sure why `shape` takes the form is does in lines 468---471 as a result of this boolean attribute.

Reply:
Yes. Although it turns out I don't actually need this property here.

Reply:
Removed.